### PR TITLE
Implement leap motion hand tracking

### PR DESCRIPTION
**Pull request by FejZa**
# XRTK - Mixed Reality Toolkit Pull Request

## Overview

Adds hand tracking support for using the leap motion sensor.
Blocked by https://github.com/XRTK/XRTK-Core/pull/552

### CODE REVIEW

**Review by StephenHodgson — CHANGES_REQUESTED** [review]
I really love the enthusiasm, but I think I wanna handle the plugins a bit differently. 
Can I take a stab at a first pass for this?

**Comment by StephenHodgson** [comment]
Mainly check out the work I'm doing in the SteamVR and Lumin packages respectively

For the most part we're just copying the files and keeping them in the ignored file list. We we're not checking in any of that stuff directly into our repos

**Comment by FejZa** [comment]
This is a work in progress Stephen.

**Comment by FejZa** [comment]
I checked those in by accident. Will remove them.

**Comment by FejZa** [comment]
It's cleaned up now :). Only the actually hand related and XRTK code left.

File: XRTK.Ultraleap/Packages/com.xrtk.ultraleap/Runtime/Extensions/QuaternionExtensions.cs [hunk 1]
```diff
@@ -14,7 +14,14 @@ public static class QuaternionExtensions
...
-        /// <returns>A <see cref="Quaternion"/>.</returns>
+        /// <returns>A <see cref="Quaternion"/> with identical values to the input <see cref="Leap.LeapQuaternion"/>.</returns>
         public static Quaternion ToQuaternion(this Leap.LeapQuaternion q) => new Quaternion(q.x, q.y, q.z, q.w);
+
+        /// <summary>
+        /// Converts a <see cref="LeapInternal.LEAP_QUATERNION"/> object to a <see cref="Quaternion"/> object.
+        /// </summary>
+        /// <param name="vector">The <see cref="LeapInternal.LEAP_QUATERNION"/> to convert.</param>
+        /// <returns>A <see cref="Quaternion"/> with identical values to the input <see cref="LeapInternal.LEAP_QUATERNION"/>.</returns>
+        public static Quaternion ToQuaternion(this LeapInternal.LEAP_QUATERNION q) => new Quaternion(q.x, q.y, q.z, q.w);
```
Comment (StephenHodgson) [inline XRTK.Ultraleap/Packages/com.xrtk.ultraleap/Runtime/Extensions/QuaternionExtensions.cs]: I think it'd be better to do an implicit cast for these. Check out what I did in the Lumin package. You could update it to implicitly cast from `LeapInternal.LEAP_QUATERNION` to `Quaternion`

**Comment by StephenHodgson** [comment]
@FejZa what's left for this PR?

**Review by StephenHodgson — APPROVED** [review]
Works for me

**Pull request merged**